src/Models/BaseElement.php

```diff
@@ -528,6 +538,59 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search if ElementSiteTreeFilterSearch.render_elements is false
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $fieldNames = $this->getTextualDatabaseFieldNames();
+        $excludedFieldNames = $this->getFieldNamesExcludedFromCmsSearch();
+        $contents = [];
+        foreach ($fieldNames as $fieldName) {
+            if (in_array($fieldName, $excludedFieldNames)) {
+                continue;
+            }
+            $contents[] = $this->$fieldName;
+        }
+        // Allow projects to update contents of third-party elements.
+        $this->extend('updateContentForCmsSearch', $contents);
+        
+        // Use |#| to delimit different fields rather than space so that you don't
+        // accidentally join results of two columns that are next to each other in a table
+        $content = implode('|#|', array_filter($contents));
```

Comment:
Should we use `|%|` or `|#|` consistently everywhere? And is this already a good enough example that it would make sense to have it either as a config or at least as a class constant so it's consistent across usages?

Reply:
It really makes no difference, I just used different delimeter in different places so that it's easier to debug where things are coming from

Reply:
Yeah I don't think it matters so much _what_ the delimiter is - all that matters is that there _is_ a delimiter, and that the delimiter isn't something likely to be included as a search term.